### ISSUE TITLE
BasicAuthentication should decode credentials as UTF-8

### ISSUE THREAD

**Issue by kennell** [encode/django-rest-framework#7183]
## Checklist

- [X] I have verified that that issue exists against the `master` branch of Django REST framework.
- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [X] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [X] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

1. Create a simple view that is protected by `BasicAuthentication`
2. Create a `User` with username `walterwhite` and password `pässwörd`
3. Perform a request against the protected view with valid credentials in the HTTP `Authorization` header, e.g. in this case `Basic d2FsdGVyd2hpdGU6cMOkc3N3w7ZyZA==`

## Expected behavior

HTTP 200 OK

## Actual behavior

HTTP 401 Unauthorized

## Analysis

This all boils down to [this line](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authentication.py#L77) in the BasicAuthentication auth backend. Basically any username/password combination that contains non-`iso-8859-1` characters will lead to a authentication failing even if those credentials are correct. The example credentials mentinoned above are decoded as `('walterwhite', ':', 'pÃ¤sswÃ¶rd')` which leads to `authenticate()` failing and subsequently raising HTTP 401.

[RFC5987](https://tools.ietf.org/html/rfc5987) indeed specifies that HTTP header key-value pairs  should be encoded/decoded using `iso-8859-1` and this is fine since the **value** of the Authorization Header is a base64 string and will only contains ASCII characters. However, there is no reason why we need to use `iso-8859-1` when decoding that base64 string to receive the credentials.

tl;dr: i suggest using `utf-8` when decoding the `Authorization` header credentials.

**Comment by kevin-brown** [comment]
I'm willing to entertain a pull request with this change. This makes sense, and while I understand why it was done the way it was originally, it makes sense to use UTF-8 instead.

**Comment by tomchristie** [comment]
I'd probably suggest decoding as `utf-8`, and falling back to `latin-1` on decode failures?

That'd be a nice robust approach, since latin-1 covers a full 8-bit character set, and won't ever raise decoding errors.

**Comment by kennell** [comment]
Fixed with https://github.com/encode/django-rest-framework/pull/7193
